rules/spec.yaml

```diff
@@ -0,0 +1,98 @@
+openapi: 3.0.0
+info:
+  description: The Rules Backend API
+  version: "0.0.1"
+  title: Rules Backend API
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+tags:
+  - name: rulesv1
+    description: Calls related to rules
+paths:
+  /api/v1/{tenant}/rules:
+    parameters:
+      - in: path
+        name: tenant
+        description: name of the tenant
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - rulesv1
+      summary: lists all rules for a tenant
+      operationId: listRules
+      description: |
+        You can list all rules for a tenant
+      responses:
+        '200':
+          description: all rules for that tenant
+          content:
+            application/yaml:
+              schema:
+                $ref: '#/components/schemas/Rules'
+        '404':
+          description: invalid tenant
+    put:
+      tags:
+        - rulesv1
+      summary: set/overwrite the rules for the tenant
+      operationId: setRules
+      description: Set rules for the tenant
+      responses:
+        '201':
+          description: rule file created
+        '400':
+          description: 'invalid rules object'
+      requestBody:
+        content:
+          application/yaml:
+            schema:
+              $ref: '#/components/schemas/Rules'
+        description: Rules to set
+components:
+  schemas:
+    Rules:
+      type: object
+      required:
+        - groups
+      properties:
+        groups:
+          type: array
+          items:
+            $ref: '#/components/schemas/RuleGroup'
+    RuleGroup:
+      type: object
+      required:
+        - interval
+        - name
+        - rules
+      properties:
+        interval:
+          type: string
+          example: '1m'
+        name:
+          type: string
+          example: 'telemeter.rules'
+        rules:
+          type: array
+          items:
+            $ref: '#/components/schemas/Rule'
+    Rule:
```

Comment:
Ok, I think the problem here is that we want to return alerting rules too. So we might want to have some union here.

Also ideally this is exactly the same as Prometheus Rules HTTP API, which has some specifics... ): 

We tried to improve this on Thanos by using protobuf and this did not pan-out really: https://thanos.io/tip/proposals-accepted/202107-protobuf-openapi-httpapi.md/